docs/index.bs

```diff
@@ -546,146 +532,112 @@ spec: rfc7231; urlPrefix: https://tools.ietf.org/html/rfc7231
+        1. Let |promise| be a new <a>promise</a>.
+        1. Run the following substeps <a>in parallel</a>:
+            1. Let |registration| be the result of running <a>Match Service Worker Registration</a> algorithm with |clientURL| as its argument.
+            1. If |registration| is not null, then:
+                1. Resolve |promise| with the {{ServiceWorkerRegistration}} object which represents |registration|.
+            1. Else:
+                1. Resolve |promise| with undefined.
+        1. Return |promise|.
     </section>
 
     <section algorithm="navigator-service-worker-getRegistrations">
       <h4 id="navigator-service-worker-getRegistrations">{{ServiceWorkerContainer/getRegistrations()}}</h4>
 
-      <p><dfn method for="ServiceWorkerContainer"><code>getRegistrations()</code></dfn> method <em class="rfc2119" title="MUST">must</em> run these steps:</p>
-
-      <ol>
-        <li>Let <var>client</var> be the <a>context object</a>'s <a for="ServiceWorkerContainer">service worker client</a>.</li>
-        <li>Let <var>promise</var> be a new <a>promise</a>.</li>
-        <li>Run the following substeps <a>in parallel</a>:
-          <ol>
-            <li>Let <var>array</var> be an empty array.</li>
-            <li>For each <a>Record</a> {\[[key]], \[[value]]} <var>entry</var> of <a href="#dfn-scope-to-registration-map">scope to registration map</a>:
-              <ol>
-                <li>If the <a for="resource">origin</a> of the result of <a lt="URL parser">parsing</a> <var>entry</var>.\[[key]] is the <a lt="same origin">same</a> as <var>client</var>'s <a for="resource">origin</a>, and <var>entry</var>.\[[value]]'s <a href="#dfn-uninstalling-flag">uninstalling flag</a> is unset, add the {{ServiceWorkerRegistration}} object associated with <var>entry</var>.\[[value]] to the <var>array</var>.</li>
-              </ol>
-            </li>
-            <li>Resolve <var>promise</var> with <var>array</var>.</li>
-          </ol>
-        </li>
-        <li>Return <var>promise</var>.</li>
-      </ol>
+      <dfn method for="ServiceWorkerContainer"><code>getRegistrations()</code></dfn> method *must* run these steps:
+
+        1. Let |client| be the <a>context object</a>'s <a for="ServiceWorkerContainer">service worker client</a>.
+        1. Let |promise| be a new <a>promise</a>.
+        1. Run the following substeps <a>in parallel</a>:
+            1. Let |array| be an empty array.
+            1. <a for="map">For each</a> |key| → |value| of <a>scope to registration map</a>:
+                1. If the <a for="environment settings object">origin</a> of the result of <a lt="URL parser">parsing</a> |key| is the <a lt="same origin">same</a> as |client|'s <a for="environment settings object">origin</a>, and |value|'s <a>uninstalling flag</a> is unset, add the {{ServiceWorkerRegistration}} object associated with |value| to the |array|.
```

Comment:
The first origin here should be `<a for="url">` as it's referring to the origin of a url rather than that of a settings object (I know, I made quite a mess about linking to the correct origin definition whenever an origin was used).